aws_lambda_powertools/utilities/parser/envelopes/sns.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+from ..models import SnsModel
+from ..types import Model
+from .base import BaseEnvelope
+
+logger = logging.getLogger(__name__)
+
+
+class SnsEnvelope(BaseEnvelope):
+    """SNS Envelope to extract array of Records
+
+    The record's body parameter is a string, though it can also be a JSON encoded string.
+    Regardless of its type it'll be parsed into a BaseModel object.
+
+    Note: Records will be parsed the same way so if model is str,
+    all items in the list will be parsed as str and npt as JSON (and vice versa)
+    """
+
+    def parse(self, data: Optional[Union[Dict[str, Any], Any]], model: Model) -> List[Optional[Model]]:
+        """Parses records found with model provided
+
+        Parameters
+        ----------
+        data : Dict
+            Lambda event to be parsed
+        model : Model
+            Data model provided to parse after extracting data using envelope
+
+        Returns
+        -------
+        List
+            List of records parsed with model provided
+        """
+        logger.debug(f"Parsing incoming data with SNS model {SnsModel}")
+        parsed_envelope = SnsModel.parse_obj(data)
+        output = []
+        logger.debug(f"Parsing SNS records in `body` with {model}")
+        for record in parsed_envelope.Records:
+            output.append(self._parse(data=record.Sns.Message, model=model))
```

Comment:
Getting confirmation from SNS team whether there could be more than one record when using SNS as an event source for Lambda. 

SNS FAQ says otherwise despite the Records schema being an actual array - I'll revert back with an approval once I have it.

```
Q: Will a notification contain more than one message?
No, all notification messages will contain a single published message.
```

Reply:
Resolving it - SNS confirmed that they might change this in the future, let's keep the implementation as-is on the record array.